src/Infrastructure/Services/ParticipantService.cs

```diff
@@ -0,0 +1,399 @@
+        var participant = await _participantRepository.GetParticipantDetailsAsync(request.NhsId);
+        if (participant != null)
+        {
+            participant.Email = request.Email.ToLower();
+            participant.Firstname = request.Firstname;
+            participant.Lastname = request.Lastname;
+            participant.DateOfBirth = request.DateOfBirth;
+            participant.NhsNumber = request.NhsNumber;
+            participant.UpdatedAtUtc = _clock.Now();
+
+            await _participantRepository.UpdateParticipantDetailsAsync(participant);
+            return;
+        }
+
+        participant = await GetParticipantDetailsByNhsNumber(request.NhsNumber);
+        // If participant is found, create new user and deactivate old user
+        if (participant != null)
+        {
+            await CreateUserAndDeactivateOldUserAsync(request, participant);
+        }
+
+        // Look up participant by email and date of birth
+        var emailRequestId = await AdminGetUserAsync(request.Email);
+        if (emailRequestId == null) return;
+        participant = await _participantRepository.GetParticipantDetailsAsync(emailRequestId);
+        if (participant == null) return;
+        // check if string version of date of birth matches without time
+        if (participant.DateOfBirth.HasValue && request.DateOfBirth.HasValue &&
+            participant.DateOfBirth.Value.Date ==
+            request.DateOfBirth.Value.Date)
+        {
+            await CreateUserAndDeactivateOldUserAsync(request, participant);
+        }
+        else
+        {
+            // pass back error message to be displayed
+            throw new ConflictException(ErrorCode.UnableToMatchAccounts);
+        }
+    }
+
+    public async Task UpdateParticipantEmail(string participantId, string newEmail)
+    {
+        var participant = await _participantRepository.GetParticipantDetailsAsync(participantId);
+        if (participant == null)
+            throw new NotFoundException($"No participant found for participantId: {participantId}");
+
+        participant.Email = newEmail.ToLower();
+        participant.UpdatedAtUtc = _clock.Now();
+        await _participantRepository.UpdateParticipantDetailsAsync(participant);
+    }
+
+    public async Task DeleteUserAsync(string participantId)
+    {
+        try
+        {
+            var entity = await _participantRepository.GetParticipantDetailsAsync(participantId);
+            if (entity == null) return;
+
+            var linkedEmail = entity.Email;
+            await SaveAnonymisedDemographicParticipantData(entity);
+            await RemoveParticipantData(entity);
+                    
+
+            var linkedEntity = await GetParticipantDetailsByEmail(linkedEmail);
+            if (linkedEntity == null) return;
+            await RemoveParticipantData(linkedEntity);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError("Delete-error = {EMessage}", e.Message);
```

Comment:
We need the stack trace here too. 
Always pass the exception instance to the logging method where possible.
(Applies to all exception logging)

Use this overload:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logerror?view=dotnet-plat-ext-7.0#microsoft-extensions-logging-loggerextensions-logerror(microsoft-extensions-logging-ilogger-system-exception-system-string-system-object())